### ISSUE TITLE
[android] URGENT -> cannot build app: <anonymous expo.modules.adapters.reanimated.EXReanimatedPackage$1> is not abstract and does not override abstract method getBundleAssetName(boolean)

### ISSUE THREAD

**Issue by mwegener-com** [software-mansion/react-native-reanimated#2701]
## Description
I cannot build the android version of my app since updating to the newest version of Reanimated.
I'm using expo dev client.

### Expected behavior
Installation successful

###  steps to reproduce
delete android dir
expo run:android --device

## Error
```
> Task :react-native-reanimated:compileDebugJavaWithJavac FAILED
/Users/marius/Projekte/cabernet-und-co/node_modules/react-native-reanimated/android/expo/src/main/java/expo/modules/adapters/reanimated/EXReanimatedPackage.java:18: error: <anonymous expo.modules.adapters.reanimated.EXReanimatedPackage$1> is not abstract and does not override abstract method getBundleAssetName(boolean) in expo.modules.core.interfaces.ReactNativeHostHandler
    final ReactNativeHostHandler handler = new ReactNativeHostHandler() {
                                                                        ^
1 error

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':react-native-reanimated:compileDebugJavaWithJavac'.
> Compilation failed; see the compiler error output for details.

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Get more help at https://help.gradle.org

Deprecated Gradle features were used in this build, making it incompatible with Gradle 7.0.
Use '--warning-mode all' to show the individual deprecation warnings.
See https://docs.gradle.org/6.9/userguide/command_line_interface.html#sec:command_line_warnings

BUILD FAILED in 33s
451 actionable tasks: 451 executed
/Users/marius/Projekte/cabernet-und-co/android/gradlew exited with non-zero code: 1
Error: /Users/marius/Projekte/cabernet-und-co/android/gradlew exited with non-zero code: 1
    at ChildProcess.completionListener (/usr/local/lib/node_modules/expo-cli/node_modules/@expo/spawn-async/src/spawnAsync.ts:65:13)
    at Object.onceWrapper (node:events:434:26)
    at ChildProcess.emit (node:events:327:20)
    at maybeClose (node:internal/child_process:1048:16)
    at Process.ChildProcess._handle.onexit (node:internal/child_process:288:5)
    ...
    at spawnAsync (/usr/local/lib/node_modules/expo-cli/node_modules/@expo/spawn-async/src/spawnAsync.ts:26:19)
    at spawnGradleAsync (/usr/local/lib/node_modules/expo-cli/src/commands/run/android/spawnGradleAsync.ts:83:18)
    at assembleAsync (/usr/local/lib/node_modules/expo-cli/src/commands/run/android/spawnGradleAsync.ts:57:16)
    at actionAsync (/usr/local/lib/node_modules/expo-cli/src/commands/run/android/runAndroid.ts:143:9)
```

## Package versions
- React Native: 0.64.3
- React Native Reanimated: 2.3.0
- Expo: 43

## Affected platforms

- [X] Android
- [ ] iOS
- [ ] Web

**Comment by Kudo** [comment]
thanks for reporting this issue. we are working on it and provide a fix in `expo-modules-core`. will let you know once we fixed this.

**Comment by mwegener-com** [comment]
Thanks a lot!

In the meantime I tried to redo the changes as mentioned here: https://github.com/software-mansion/react-native-reanimated/pull/2690
No I'm able to build my app,  but I crashes on startup.

```
2021-12-08 02:12:19.597 8791-8791/com.mwegener.cabernetco E/unknown:ReactNative: Unable to launch logbox because react was unable to create the root view
2021-12-08 02:12:19.952 8791-8791/com.mwegener.cabernetco E/unknown:ReactInstanceManager: destroyRootView called
2021-12-08 02:12:19.952 8791-8791/com.mwegener.cabernetco E/unknown:ReactInstanceManager: destroyRootView called, unmountReactApplication
2021-12-08 02:12:20.364 8791-8880/com.mwegener.cabernetco E/unknown:ReactContextBaseJavaModule: Unhandled SoftException
    java.lang.RuntimeException: Catalyst Instance has already disappeared: requested by BlobModule
        at com.facebook.react.bridge.ReactContextBaseJavaModule.getReactApplicationContextIfActiveOrWarn(ReactContextBaseJavaModule.java:67)
        at com.facebook.react.modules.blob.BlobModule.addNetworkingHandler(BlobModule.java:300)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.facebook.react.bridge.JavaMethodWrapper.invoke(JavaMethodWrapper.java:372)
        at com.facebook.react.bridge.JavaModuleWrapper.invoke(JavaModuleWrapper.java:151)
        at com.facebook.react.bridge.queue.NativeRunnable.run(Native Method)
        at android.os.Handler.handleCallback(Handler.java:938)
        at android.os.Handler.dispatchMessage(Handler.java:99)
        at com.facebook.react.bridge.queue.MessageQueueThreadHandler.dispatchMessage(MessageQueueThreadHandler.java:27)
        at android.os.Looper.loop(Looper.java:223)
        at com.facebook.react.bridge.queue.MessageQueueThreadImpl$4.run(MessageQueueThreadImpl.java:226)
        at java.lang.Thread.run(Thread.java:923)
2021-12-08 02:12:20.375 8791-8904/com.mwegener.cabernetco E/unknown:ReactContextBaseJavaModule: Unhandled SoftException
    java.lang.RuntimeException: Catalyst Instance has already disappeared: requested by WebSocketModule
        at com.facebook.react.bridge.ReactContextBaseJavaModule.getReactApplicationContextIfActiveOrWarn(ReactContextBaseJavaModule.java:67)
        at com.facebook.react.modules.websocket.WebSocketModule.sendEvent(WebSocketModule.java:62)
        at com.facebook.react.modules.websocket.WebSocketModule.notifyWebSocketFailed(WebSocketModule.java:340)
        at com.facebook.react.modules.websocket.WebSocketModule.access$200(WebSocketModule.java:40)
        at com.facebook.react.modules.websocket.WebSocketModule$1.onFailure(WebSocketModule.java:175)
        at okhttp3.internal.ws.RealWebSocket.failWebSocket(RealWebSocket.java:570)
        at okhttp3.internal.ws.RealWebSocket$1.onFailure(RealWebSocket.java:216)
        at okhttp3.RealCall$AsyncCall.execute(RealCall.java:180)
        at okhttp3.internal.NamedRunnable.run(NamedRunnable.java:32)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)
        at java.lang.Thread.run(Thread.java:923)
2021-12-08 02:12:21.437 8791-8880/com.mwegener.cabernetco E/unknown:ReactContextBaseJavaModule: Unhandled SoftException
    java.lang.RuntimeException: Catalyst Instance has already disappeared: requested by Timing
        at com.facebook.react.bridge.ReactContextBaseJavaModule.getReactApplicationContextIfActiveOrWarn(ReactContextBaseJavaModule.java:67)
        at com.facebook.react.modules.core.TimingModule.access$000(TimingModule.java:22)
        at com.facebook.react.modules.core.TimingModule$BridgeTimerManager.callTimers(TimingModule.java:28)
        at com.facebook.react.modules.core.JavaTimerManager.createAndMaybeCallTimer(JavaTimerManager.java:341)
        at com.facebook.react.modules.core.TimingModule.createTimer(TimingModule.java:91)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.facebook.react.bridge.JavaMethodWrapper.invoke(JavaMethodWrapper.java:372)
        at com.facebook.react.bridge.JavaModuleWrapper.invoke(JavaModuleWrapper.java:151)
        at com.facebook.react.bridge.queue.NativeRunnable.run(Native Method)
        at android.os.Handler.handleCallback(Handler.java:938)
        at android.os.Handler.dispatchMessage(Handler.java:99)
        at com.facebook.react.bridge.queue.MessageQueueThreadHandler.dispatchMessage(MessageQueueThreadHandler.java:27)
        at android.os.Looper.loop(Looper.java:223)
        at com.facebook.react.bridge.queue.MessageQueueThreadImpl$4.run(MessageQueueThreadImpl.java:226)
        at java.lang.Thread.run(Thread.java:923)
2021-12-08 02:12:21.457 8791-8907/com.mwegener.cabernetco E/unknown:ReactContextBaseJavaModule: Unhandled SoftException
    java.lang.RuntimeException: Catalyst Instance has already disappeared: requested by WebSocketModule
        at com.facebook.react.bridge.ReactContextBaseJavaModule.getReactApplicationContextIfActiveOrWarn(ReactContextBaseJavaModule.java:67)
        at com.facebook.react.modules.websocket.WebSocketModule.sendEvent(WebSocketModule.java:62)
        at com.facebook.react.modules.websocket.WebSocketModule.access$100(WebSocketModule.java:40)
        at com.facebook.react.modules.websocket.WebSocketModule$1.onOpen(WebSocketModule.java:156)
        at okhttp3.internal.ws.RealWebSocket$1.onResponse(RealWebSocket.java:208)
        at okhttp3.RealCall$AsyncCall.execute(RealCall.java:174)
        at okhttp3.internal.NamedRunnable.run(NamedRunnable.java:32)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)
        at java.lang.Thread.run(Thread.java:923)
2021-12-08 02:12:21.498 8791-8880/com.mwegener.cabernetco E/unknown:ReactContextBaseJavaModule: Unhandled SoftException
    java.lang.RuntimeException: Catalyst Instance has already disappeared: requested by Networking
        at com.facebook.react.bridge.ReactContextBaseJavaModule.getReactApplicationContextIfActiveOrWarn(ReactContextBaseJavaModule.java:67)
        at com.facebook.react.modules.network.NetworkingModule.getEventEmitter(NetworkingModule.java:767)
        at com.facebook.react.modules.network.NetworkingModule.sendRequestInternal(NetworkingModule.java:273)
        at com.facebook.react.modules.network.NetworkingModule.sendRequest(NetworkingModule.java:243)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.facebook.react.bridge.JavaMethodWrapper.invoke(JavaMethodWrapper.java:372)
        at com.facebook.react.bridge.JavaModuleWrapper.invoke(JavaModuleWrapper.java:151)
        at com.facebook.react.bridge.queue.NativeRunnable.run(Native Method)
        at android.os.Handler.handleCallback(Handler.java:938)
        at android.os.Handler.dispatchMessage(Handler.java:99)
        at com.facebook.react.bridge.queue.MessageQueueThreadHandler.dispatchMessage(MessageQueueThreadHandler.java:27)
        at android.os.Looper.loop(Looper.java:223)
        at com.facebook.react.bridge.queue.MessageQueueThreadImpl$4.run(MessageQueueThreadImpl.java:226)
        at java.lang.Thread.run(Thread.java:923)
2021-12-08 02:12:21.516 8791-8791/com.mwegener.cabernetco E/unknown:ReactNative: Unable to handle Exception - catalystInstanceVariableExists: true - isCatalystInstanceAlive: true - hasExceptionHandler: true
    java.lang.NullPointerException: Attempt to invoke interface method 'boolean com.swmansion.reanimated.layoutReanimation.NativeMethodsHolder.isLayoutAnimationEnabled()' on a null object reference
        at com.swmansion.reanimated.layoutReanimation.AnimationsManager.isLayoutAnimationEnabled(AnimationsManager.java:507)
        at com.swmansion.reanimated.layoutReanimation.ReaLayoutAnimator.isLayoutAnimationEnabled(ReanimatedNativeHierarchyManager.java:191)
        at com.swmansion.reanimated.layoutReanimation.ReanimatedNativeHierarchyManager.manageChildren(ReanimatedNativeHierarchyManager.java:266)
        at com.facebook.react.uimanager.UIViewOperationQueue$ManageChildrenOperation.execute(UIViewOperationQueue.java:211)
        at com.facebook.react.uimanager.UIViewOperationQueue$1.run(UIViewOperationQueue.java:908)
        at com.facebook.react.uimanager.UIViewOperationQueue.flushPendingBatches(UIViewOperationQueue.java:1019)
        at com.facebook.react.uimanager.UIViewOperationQueue.access$2600(UIViewOperationQueue.java:47)
        at com.facebook.react.uimanager.UIViewOperationQueue$2.runGuarded(UIViewOperationQueue.java:977)
        at com.facebook.react.bridge.GuardedRunnable.run(GuardedRunnable.java:31)
        at android.os.Handler.handleCallback(Handler.java:938)
        at android.os.Handler.dispatchMessage(Handler.java:99)
        at android.os.Looper.loop(Looper.java:223)
        at android.app.ActivityThread.main(ActivityThread.java:7523)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:592)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:941)
2021-12-08 02:12:21.518 8791-8791/com.mwegener.cabernetco E/AndroidRuntime: FATAL EXCEPTION: main
    Process: com.mwegener.cabernetco, PID: 8791
    java.lang.IllegalStateException: java.lang.NullPointerException: Attempt to invoke interface method 'boolean com.swmansion.reanimated.layoutReanimation.NativeMethodsHolder.isLayoutAnimationEnabled()' on a null object reference
        at com.facebook.react.bridge.ReactContext.handleException(ReactContext.java:387)
        at com.facebook.react.bridge.ReactContext$ExceptionHandlerWrapper.handleException(ReactContext.java:394)
        at com.facebook.react.bridge.GuardedRunnable.run(GuardedRunnable.java:33)
        at android.os.Handler.handleCallback(Handler.java:938)
        at android.os.Handler.dispatchMessage(Handler.java:99)
        at android.os.Looper.loop(Looper.java:223)
        at android.app.ActivityThread.main(ActivityThread.java:7523)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:592)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:941)
     Caused by: java.lang.NullPointerException: Attempt to invoke interface method 'boolean com.swmansion.reanimated.layoutReanimation.NativeMethodsHolder.isLayoutAnimationEnabled()' on a null object reference
        at com.swmansion.reanimated.layoutReanimation.AnimationsManager.isLayoutAnimationEnabled(AnimationsManager.java:507)
        at com.swmansion.reanimated.layoutReanimation.ReaLayoutAnimator.isLayoutAnimationEnabled(ReanimatedNativeHierarchyManager.java:191)
        at com.swmansion.reanimated.layoutReanimation.ReanimatedNativeHierarchyManager.manageChildren(ReanimatedNativeHierarchyManager.java:266)
        at com.facebook.react.uimanager.UIViewOperationQueue$ManageChildrenOperation.execute(UIViewOperationQueue.java:211)
        at com.facebook.react.uimanager.UIViewOperationQueue$1.run(UIViewOperationQueue.java:908)
        at com.facebook.react.uimanager.UIViewOperationQueue.flushPendingBatches(UIViewOperationQueue.java:1019)
        at com.facebook.react.uimanager.UIViewOperationQueue.access$2600(UIViewOperationQueue.java:47)
        at com.facebook.react.uimanager.UIViewOperationQueue$2.runGuarded(UIViewOperationQueue.java:977)
        at com.facebook.react.bridge.GuardedRunnable.run(GuardedRunnable.java:31)
        at android.os.Handler.handleCallback(Handler.java:938) 
        at android.os.Handler.dispatchMessage(Handler.java:99) 
        at android.os.Looper.loop(Looper.java:223) 
        at android.app.ActivityThread.main(ActivityThread.java:7523) 
        at java.lang.reflect.Method.invoke(Native Method) 
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:592) 
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:941) 
```

**Comment by Kudo** [comment]
we've published `expo@43.0.4` and `expo-modules-core@0.4.9` for the `EXReanimatedPackage is not abstract and does not override abstract method` issue. you could probably run `yarn upgrade expo` to upgrade both. if this doesn't works for you, please share your `yarn why expo-modules-core` output with us. thanks!

for `Attempt to invoke interface method 'boolean com.swmansion.reanimated.layoutReanimation.NativeMethodsHolder.isLayoutAnimationEnabled()' on a null object reference` issue, it seems to be another problem and i cannot reproduce locally. maybe you can create a new issue with [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example). reanimated team or we will happy to help with that.

**Comment by mwegener-com** [comment]
`yarn upgrade expo` went fine. Other problem this exists, I will create a new Issue for that.
